### ISSUE TITLE
Classes as implementations of protocol

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#2729]
**Is your feature request related to a problem? Please describe.**

It should be possible to pass a class as an instance of a protocol, given that the class object fulfils the protocol.

The same issue on mypy, with reproducer: https://github.com/python/mypy/issues/4536
Copying here for recording purposes:
```python
from typing_extensions import Protocol

class Jumps(Protocol):
    def jump(self) -> int:
        ...

class Jumper1:
    @classmethod
    def jump(cls) -> int:
        print("class jumping")
        return 1

class Jumper2:
    def jump(self) -> int:
        print("instance jumping")
        return 2

def do_jump(j: Jumps):
    print(j.jump())

do_jump(Jumper1)
do_jump(Jumper2())
```

Basically the same as https://github.com/microsoft/pyright/issues/1341 but for class objects.

PEP 544 doesn't seem to mention this explicitly, but I don't know any arguments against.

**Describe the solution you'd like**
The reproducer should typecheck.

**Comment by erictraut** [comment]
No, this isn't allowed. The semantics of a class method and an instance method are different, especially when they're invoked from the class itself.

You can use a classmethod in your protocol definition if you would like.

```python
class Jumps(Protocol):
    @classmethod
    def jump(cls) -> int:
        ...
```

**Comment by matejcik** [comment]
How is this different from accepting modules then?

For module compatibility, you have to drop the `self` argument. The semantics are also different.
I suppose I _could_ abuse the passed-in instance hard enough that a class object would fail but class instance would work, but I can't imagine that a module object would work in the same case.

Even if for some reason classmethods should be included, `@staticmethod`s should be accepted.

**Comment by erictraut** [comment]
Modules were carved out as a special case in PEP 544. Class methods were not. If you think this is an important scenario to support, please work with the typing community to amend the PEP.

**Comment by matejcik** [comment]
it turns out that PEP 544 says something about this: https://www.python.org/dev/peps/pep-0544/#type-and-class-objects-vs-protocols
> A class object is considered an implementation of a protocol if accessing all members on it results in types compatible with the protocol members. For example:
```python
from typing import Any, Protocol

class ProtoA(Protocol):
    def meth(self, x: int) -> int: ...
class ProtoB(Protocol):
    def meth(self, obj: Any, x: int) -> int: ...

class C:
    def meth(self, x: int) -> int: ...

a: ProtoA = C  # Type check error, signatures don't match!
b: ProtoB = C  # OK
```

Their example is a little weird, but it seems to suggest that if the _revealed type_ of `instance.foo` is identical to revealed type of `Class.foo`, then `Class` should be considered as implementing the same protocol as `instance`.

**Comment by erictraut** [comment]
Oh, good call. I missed that part of the PEP.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.198, which I just published. It will also be included in the next release of pylance.
